ts/main.ts

```diff
@@ -328,10 +340,25 @@ setStatsText();
 function setStatsText(): void {
   $guessesTxt.textContent = guesses.toString();
   $correctTxt.textContent = numCorrect.toString();
+
   if (!(numCorrect / guesses)) {
-    $accuracyTxt.textContent = '100%';
+    $accuracyTxt.textContent = '0%';
   } else {
-    $accuracyTxt.textContent = (numCorrect / guesses).toString() + '%';
+    $accuracyTxt.textContent =
+      Math.floor((numCorrect / guesses) * 100).toString() + '%';
+  }
+
+  if (completedSquares === gridSize * gridSize && $statsModal) {
+    $statsModal.classList.remove('hidden');
+    const $statsHeading = document.querySelector(
+      '.stats-heading',
+    ) as HTMLHeadingElement;
+    if (!$statsHeading) throw new Error('.stats-heading query failed');
+    if (numCorrect / guesses === 1) {
+      $statsHeading.textContent = 'Perfect!';
+    } else if (numCorrect / guesses >= 0.5) {
+      $statsHeading.textContent = 'Nice Work!';
+    }
```

Comment:
What happens if the user does not get 'Perfect' or 'Nice work'? If you don't have a default at the end then would the last message be shown such as user getting a 0% and it still says 'Perfect!' from before?